frappe/desk/query_report.py

```diff
@@ -240,6 +247,11 @@ def add_custom_column_data(custom_columns, result):
 				# possible if the row is empty
 				if not row_reference:
 					continue
+				if (
+					custom_column_data.get(key).get(row_reference) is None
+					and key[0] in doctype_name_from_custom_field
+				):
+					column["fieldname"] = column.get("fieldname") + "-" + frappe.unscrub(key[0])
```

Comment:
@RitvikSardana please add a test case for this use case of multiple fields with same name. Also in the description explain how this fixes the problem (i.e. implementation)

Observation:
Ideally, each field should be uniquely identifiable by `tablename.fieldname`.